zkevm-circuits/src/witness/tx.rs

```diff
@@ -389,12 +389,40 @@ impl Transaction {
             byte_idx: 0,
             depth: 0,
         };
+
+        // Queue up stack operations
+        let mut stack_ops: Vec<RlpStackOp<F>> = vec![];
+        let mut last_bytes_on_depth: [usize; 4] = [0, 0, 0, 0];
+        let mut stack_acc = Value::known(F::zero());
+
+        let mut stack_acc_pow_of_rand: Vec<Value<F>> = vec![];
+        let mut pow_of_rand = Value::known(F::one());
+        stack_acc_pow_of_rand.push(pow_of_rand);
+        for _ in 0..4 {
+            pow_of_rand = pow_of_rand * keccak_rand;
+            stack_acc_pow_of_rand.push(pow_of_rand);
+        }
```

Comment:
We can refactor this into something like this:

```suggestion
        let stack_acc_pow_of_rand = std::iter::successors(Some(Value::known(F::one())), |coeff| {
        Some(keccak_rand * coeff)
        })
        .take(4)
        .collect::<Vec<Value<F>>>();
```

Reply:
Refactored.